lib/srv/desktop/tdp/proto.go

```diff
@@ -242,88 +216,32 @@ func decodePNGFrame(in byteReader) (PNGFrame, error) {
 	return PNGFrame{Img: img}, nil
 }
 
-// PNG2Frame is a newer versin of PNGFrame that includes the
+// PNG2Frame is a newer version of PNGFrame that includes the
 // length of the PNG data.
-type PNG2Frame PNGFrame
-
-func decodePNG2Frame(in byteReader) (PNG2Frame, error) {
-	var length uint32
-	var header struct {
-		Left, Top     uint32
-		Right, Bottom uint32
-	}
-	if err := binary.Read(in, binary.BigEndian, &length); err != nil {
-		return PNG2Frame{}, trace.Wrap(err)
-	}
-	if err := binary.Read(in, binary.BigEndian, &header); err != nil {
-		return PNG2Frame{}, trace.Wrap(err)
-	}
-	img, err := png.Decode(io.LimitReader(in, int64(length)))
-	if err != nil {
-		return PNG2Frame{}, trace.Wrap(err)
-	}
-	// PNG encoding does not preserve offset image bounds.
-	// Opportunistically restore them based on the header.
-	switch img := img.(type) {
-	case *image.RGBA:
-		img.Rect = image.Rect(int(header.Left), int(header.Top), int(header.Right), int(header.Bottom))
-	case *image.NRGBA:
-		img.Rect = image.Rect(int(header.Left), int(header.Top), int(header.Right), int(header.Bottom))
-	}
-	return PNG2Frame{Img: img}, nil
+type PNG2Frame struct {
+	// Msg is already encoded PNG2Frame message
+	Msg []byte
 }
 
-func readRawPNG2Frame(firstByte byte, in byteReader) ([]byte, error) {
-	if MessageType(firstByte) != TypePNG2Frame {
-		return nil, trace.Wrap(trace.BadParameter("incorrect first byte passed to readRawPNG2Frame, expected %v but got %v", TypePNG2Frame, firstByte))
-	}
-
+func decodePNG2Frame(in byteReader) (PNG2Frame, error) {
 	var pngLength uint32
 	if err := binary.Read(in, binary.BigEndian, &pngLength); err != nil {
-		return nil, trace.Wrap(err)
+		return PNG2Frame{}, trace.Wrap(err)
 	}
 
 	b := make([]byte, 1+4+pngLength+16)
```

Comment:
nit: add a comment explaining what these numbers mean. I understand first byte is the type, next 4 bytes are the length, but I don't know what the `+16` is for.